
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://PlasmoGenEpi.github.io/portable-microhaplotype-object/">
      
      
      
        <link rel="next" href="about/">
      
      
      <link rel="icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.15">
    
    
      
        <title>portable-microhaplotype-object</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="css/ansi-colours.css">
    
      <link rel="stylesheet" href="css/jupyter-cells.css">
    
      <link rel="stylesheet" href="css/pandas-dataframe.css">
    
    <script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#portable-microhaplotype-object" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="." title="portable-microhaplotype-object" class="md-header__button md-logo" aria-label="portable-microhaplotype-object" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            portable-microhaplotype-object
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Index
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/PlasmoGenEpi/portable-microhaplotype-object" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="." title="portable-microhaplotype-object" class="md-nav__button md-logo" aria-label="portable-microhaplotype-object" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    portable-microhaplotype-object
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/PlasmoGenEpi/portable-microhaplotype-object" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Index
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="." class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Index
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      Classes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      Slots
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      Enumerations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    <span class="md-ellipsis">
      Types
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      Subsets
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="about/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    About
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      Classes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      Slots
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      Enumerations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    <span class="md-ellipsis">
      Types
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      Subsets
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="portable-microhaplotype-object">portable-microhaplotype-object</h1>
<p>A schema to define the minimum amount of data needed to export a microhaplotype calling pipeline analysis with associated metadata</p>
<p>URI: https://plasmogenepi.github.io/portable-microhaplotype-object</p>
<p>Name: portable-microhaplotype-object</p>
<h2 id="classes">Classes</h2>
<table>
<thead>
<tr>
<th>Class</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="BioinformaticsMethodInfo/">BioinformaticsMethodInfo</a></td>
<td>the targeted amplicon bioinformatics pipeline</td>
</tr>
<tr>
<td><a href="BioinformaticsRunInfo/">BioinformaticsRunInfo</a></td>
<td>Information about the pipeline run that generated some of the microhaplotype ...</td>
</tr>
<tr>
<td><a href="BioMethod/">BioMethod</a></td>
<td>methodology description of a portion of a bioinformatics pipeline</td>
</tr>
<tr>
<td><a href="DetectedMicrohaplotypes/">DetectedMicrohaplotypes</a></td>
<td>the microhaplotypes detected in a targeted amplicon analysis</td>
</tr>
<tr>
<td><a href="DetectedMicrohaplotypesForSample/">DetectedMicrohaplotypesForSample</a></td>
<td>Microhaplotypes detected for a sample for all targets</td>
</tr>
<tr>
<td><a href="DetectedMicrohaplotypesForTarget/">DetectedMicrohaplotypesForTarget</a></td>
<td>Microhaplotypes detected for a specific target</td>
</tr>
<tr>
<td><a href="ExperimentInfo/">ExperimentInfo</a></td>
<td>Information about a specific amplification and sequencing of a specimen</td>
</tr>
<tr>
<td><a href="GenomeInfo/">GenomeInfo</a></td>
<td>information on a genome</td>
</tr>
<tr>
<td><a href="GenomicLocation/">GenomicLocation</a></td>
<td>information on the genomic location of specific sequence</td>
</tr>
<tr>
<td><a href="MarkerOfInterest/">MarkerOfInterest</a></td>
<td>A specific genomic location of interest, e</td>
</tr>
<tr>
<td><a href="MaskingInfo/">MaskingInfo</a></td>
<td>information about a subsegment of the sequence that should be masked</td>
</tr>
<tr>
<td><a href="MicrohaplotypeForTarget/">MicrohaplotypeForTarget</a></td>
<td>Microhaplotype detected for a specific target</td>
</tr>
<tr>
<td><a href="PanelInfo/">PanelInfo</a></td>
<td>information on a panel of targeted amplicon primer pairs</td>
</tr>
<tr>
<td><a href="ParasiteDensity/">ParasiteDensity</a></td>
<td>method and value of determined parasite density</td>
</tr>
<tr>
<td><a href="PlateInfo/">PlateInfo</a></td>
<td>Information about a plate location in a standard 96 well plate</td>
</tr>
<tr>
<td><a href="PmoGenerationMethod/">PmoGenerationMethod</a></td>
<td>Information about how a PMO was generated</td>
</tr>
<tr>
<td><a href="PmoHeader/">PmoHeader</a></td>
<td>Information on the PMO file</td>
</tr>
<tr>
<td><a href="PortableMicrohaplotypeObject/">PortableMicrohaplotypeObject</a></td>
<td>Information on final results from a targeted amplicon analysis</td>
</tr>
<tr>
<td><a href="PrimerInfo/">PrimerInfo</a></td>
<td>information on a primer sequence</td>
</tr>
<tr>
<td><a href="ProjectInfo/">ProjectInfo</a></td>
<td>Information on project info</td>
</tr>
<tr>
<td><a href="ReactionInfo/">ReactionInfo</a></td>
<td>information on a panel of targeted amplicon primer pairs</td>
</tr>
<tr>
<td><a href="ReadCountsByStage/">ReadCountsByStage</a></td>
<td>Information on the reads counts at several stages of a pipeline</td>
</tr>
<tr>
<td><a href="ReadCountsByStageForExperimentalSample/">ReadCountsByStageForExperimentalSample</a></td>
<td>Information on the reads counts at several stages of a pipeline for a experim...</td>
</tr>
<tr>
<td><a href="ReadCountsByStageForTarget/">ReadCountsByStageForTarget</a></td>
<td>Information on the reads counts at several stages of a pipeline for a target</td>
</tr>
<tr>
<td><a href="RepresentativeMicrohaplotype/">RepresentativeMicrohaplotype</a></td>
<td>the representative sequence for a microhaplotype, similar to a fast(a/q) form...</td>
</tr>
<tr>
<td><a href="RepresentativeMicrohaplotypes/">RepresentativeMicrohaplotypes</a></td>
<td>a collection of representative sequences for microhaplotypes for all targets</td>
</tr>
<tr>
<td><a href="RepresentativeMicrohaplotypesForTarget/">RepresentativeMicrohaplotypesForTarget</a></td>
<td>a list of the representative sequence for a microhaplotypes, similar to a fas...</td>
</tr>
<tr>
<td><a href="SequencingInfo/">SequencingInfo</a></td>
<td>Information on sequencing info</td>
</tr>
<tr>
<td><a href="SpecimenInfo/">SpecimenInfo</a></td>
<td>Information on specimen info</td>
</tr>
<tr>
<td><a href="StageReadCounts/">StageReadCounts</a></td>
<td>Information on the reads counts at several stages</td>
</tr>
<tr>
<td><a href="TargetInfo/">TargetInfo</a></td>
<td>Information about a specific target within a genome</td>
</tr>
</tbody>
</table>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Slot</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="accession/">accession</a></td>
<td>ERA/SRA accession number for the sample if it was submitted</td>
</tr>
<tr>
<td><a href="additional_argument/">additional_argument</a></td>
<td>any additional arguments that differ from the default</td>
</tr>
<tr>
<td><a href="additional_methods/">additional_methods</a></td>
<td>any additional methods used to analyze the data</td>
</tr>
<tr>
<td><a href="alt_annotations/">alt_annotations</a></td>
<td>a list of additional annotations associated with this microhaplotype, e</td>
</tr>
<tr>
<td><a href="alternate_identifiers/">alternate_identifiers</a></td>
<td>a list of optional alternative names for the specimens</td>
</tr>
<tr>
<td><a href="associations/">associations</a></td>
<td>a list of associations with this marker, e</td>
</tr>
<tr>
<td><a href="bioinformatics_method_name/">bioinformatics_method_name</a></td>
<td>name of the collection of methods is called, e</td>
</tr>
<tr>
<td><a href="bioinformatics_methods_id/">bioinformatics_methods_id</a></td>
<td>the index into the bioinformatics_methods_info list</td>
</tr>
<tr>
<td><a href="bioinformatics_methods_info/">bioinformatics_methods_info</a></td>
<td>the bioinformatics pipeline/methods used to generated the amplicon analysis f...</td>
</tr>
<tr>
<td><a href="bioinformatics_run_id/">bioinformatics_run_id</a></td>
<td>the index into bioinformatics_run_info list</td>
</tr>
<tr>
<td><a href="bioinformatics_run_info/">bioinformatics_run_info</a></td>
<td>the runtime info for the bioinformatics pipeline used to generated the amplic...</td>
</tr>
<tr>
<td><a href="bioinformatics_run_name/">bioinformatics_run_name</a></td>
<td>a name to for this run, needs to be unique to each run</td>
</tr>
<tr>
<td><a href="BioProject_accession/">BioProject_accession</a></td>
<td>an SRA bioproject accession e</td>
</tr>
<tr>
<td><a href="chrom/">chrom</a></td>
<td>the chromosome name</td>
</tr>
<tr>
<td><a href="chromosomes/">chromosomes</a></td>
<td>a list of chromosomes found within this genome</td>
</tr>
<tr>
<td><a href="collection_country/">collection_country</a></td>
<td>the name of country collected in, would be the same as admin level 0</td>
</tr>
<tr>
<td><a href="collection_date/">collection_date</a></td>
<td>the date of the specimen collection, can be YYYY, YYYY-MM, or YYYY-MM-DD</td>
</tr>
<tr>
<td><a href="creation_date/">creation_date</a></td>
<td>the date of when the PMO file was created or modified, should be YYYY-MM-DD</td>
</tr>
<tr>
<td><a href="date_measured/">date_measured</a></td>
<td>the date the qpcr was performed, can be YYYY, YYYY-MM, or YYYY-MM-DD</td>
</tr>
<tr>
<td><a href="demultiplexing_method/">demultiplexing_method</a></td>
<td>the demultiplexing method used to separate raw reads from barcodes and primer...</td>
</tr>
<tr>
<td><a href="denoising_method/">denoising_method</a></td>
<td>the method used to de-noise and/or cluster the raw reads</td>
</tr>
<tr>
<td><a href="density_method/">density_method</a></td>
<td>the method of how this density was obtained</td>
</tr>
<tr>
<td><a href="density_method_comments/">density_method_comments</a></td>
<td>additional comments about how the density was performed</td>
</tr>
<tr>
<td><a href="detected_microhaplotypes/">detected_microhaplotypes</a></td>
<td>the microhaplotypes detected in this projects</td>
</tr>
<tr>
<td><a href="drug_usage/">drug_usage</a></td>
<td>Any drug used by subject and the frequency of usage; can include multiple dru...</td>
</tr>
<tr>
<td><a href="end/">end</a></td>
<td>the end of the location, 0-based positioning</td>
</tr>
<tr>
<td><a href="env_broad_scale/">env_broad_scale</a></td>
<td>the broad environment from which the specimen was collected, e</td>
</tr>
<tr>
<td><a href="env_local_scale/">env_local_scale</a></td>
<td>the local environment from which the specimen was collected, e</td>
</tr>
<tr>
<td><a href="env_medium/">env_medium</a></td>
<td>the environment medium from which the specimen was collected from</td>
</tr>
<tr>
<td><a href="experiment_info/">experiment_info</a></td>
<td>a list of experiments of all the seq/amp of the specimens within this project</td>
</tr>
<tr>
<td><a href="experiment_sample_id/">experiment_sample_id</a></td>
<td>the index into the experiment_info list</td>
</tr>
<tr>
<td><a href="experiment_sample_name/">experiment_sample_name</a></td>
<td>a unique identifier for this sequence/amplification run on a specimen_name</td>
</tr>
<tr>
<td><a href="experiment_samples/">experiment_samples</a></td>
<td>a list of the microhaplotypes detected for a sample by targets</td>
</tr>
<tr>
<td><a href="forward_primer/">forward_primer</a></td>
<td>the forward primer associated with this target</td>
</tr>
<tr>
<td><a href="gene_name/">gene_name</a></td>
<td>an identifier of the gene, if any, is being covered with this targeted</td>
</tr>
<tr>
<td><a href="generation_method/">generation_method</a></td>
<td>the generation method to create this PMO</td>
</tr>
<tr>
<td><a href="genome_id/">genome_id</a></td>
<td>the index to the genome in the targeted_genomes list that this location refer...</td>
</tr>
<tr>
<td><a href="genome_version/">genome_version</a></td>
<td>the genome version</td>
</tr>
<tr>
<td><a href="geo_admin1/">geo_admin1</a></td>
<td>geographical admin level 1, the secondary large demarcation of a nation (nati...</td>
</tr>
<tr>
<td><a href="geo_admin2/">geo_admin2</a></td>
<td>geographical admin level 2, the third large demarcation of a nation (nation =...</td>
</tr>
<tr>
<td><a href="geo_admin3/">geo_admin3</a></td>
<td>geographical admin level 3, the third large demarcation of a nation (nation =...</td>
</tr>
<tr>
<td><a href="gff_url/">gff_url</a></td>
<td>a link to the where this genome's annotation file could be downloaded</td>
</tr>
<tr>
<td><a href="host_age/">host_age</a></td>
<td>if specimen is from a person, the age in years of the person, can be float va...</td>
</tr>
<tr>
<td><a href="host_sex/">host_sex</a></td>
<td>if specimen is from a person, the sex listed for that person</td>
</tr>
<tr>
<td><a href="host_subject_id/">host_subject_id</a></td>
<td>an identifier for the individual a specimen was collected from</td>
</tr>
<tr>
<td><a href="host_taxon_id/">host_taxon_id</a></td>
<td>the NCBI taxonomy number of the host that the specimen was collected from</td>
</tr>
<tr>
<td><a href="insert_location/">insert_location</a></td>
<td>the intended genomic location of the insert of the amplicon (the location bet...</td>
</tr>
<tr>
<td><a href="lat_lon/">lat_lon</a></td>
<td>the latitude and longitude of the collection site of the specimen</td>
</tr>
<tr>
<td><a href="library_kit/">library_kit</a></td>
<td>Name, version, and applicable cell or cycle numbers for the kit used to prepa...</td>
</tr>
<tr>
<td><a href="library_layout/">library_layout</a></td>
<td>Specify the configuration of reads, e</td>
</tr>
<tr>
<td><a href="library_prep_plate_info/">library_prep_plate_info</a></td>
<td>plate location of where experiment was prepared for sequencing</td>
</tr>
<tr>
<td><a href="library_screen/">library_screen</a></td>
<td>Describe enrichment, screening, or normalization methods applied during ampli...</td>
</tr>
<tr>
<td><a href="library_selection/">library_selection</a></td>
<td>how amplification was done (common are PCR=Source material was selected by de...</td>
</tr>
<tr>
<td><a href="library_source/">library_source</a></td>
<td>Source of amplification material (common names GENOMIC, TRANSCRIPTOMIC)</td>
</tr>
<tr>
<td><a href="library_strategy/">library_strategy</a></td>
<td>what the nuceloacid sequencing/amplification strategy was (common names are A...</td>
</tr>
<tr>
<td><a href="location/">location</a></td>
<td>what the intended genomic location of the primer is</td>
</tr>
<tr>
<td><a href="marker_location/">marker_location</a></td>
<td>the genomic location</td>
</tr>
<tr>
<td><a href="markers_of_interest/">markers_of_interest</a></td>
<td>a list of covered markers of interest</td>
</tr>
<tr>
<td><a href="masking/">masking</a></td>
<td>masking info for the sequence</td>
</tr>
<tr>
<td><a href="mhap_id/">mhap_id</a></td>
<td>the index for a microhaplotype for a target in the representative_microhaplot...</td>
</tr>
<tr>
<td><a href="mhap_location/">mhap_location</a></td>
<td>a genomic location that was analyzed for this target info, this allows listin...</td>
</tr>
<tr>
<td><a href="mhaps/">mhaps</a></td>
<td>a list of the microhaplotypes detected for this target</td>
</tr>
<tr>
<td><a href="mhaps_target_id/">mhaps_target_id</a></td>
<td>the index for a target in the representative_microhaplotypes list</td>
</tr>
<tr>
<td><a href="microhaplotype_name/">microhaplotype_name</a></td>
<td>an optional name for this microhaplotype</td>
</tr>
<tr>
<td><a href="microhaplotypes/">microhaplotypes</a></td>
<td>a list of the microhaplotypes detected for a target</td>
</tr>
<tr>
<td><a href="microscopy_parasite_density_info/">microscopy_parasite_density_info</a></td>
<td>one or more parasite densities in microliters for this specimen</td>
</tr>
<tr>
<td><a href="name/">name</a></td>
<td>name of the genome</td>
</tr>
<tr>
<td><a href="nucl_acid_amp/">nucl_acid_amp</a></td>
<td>Link to a reference or kit that describes the enzymatic amplification of nucl...</td>
</tr>
<tr>
<td><a href="nucl_acid_amp_date/">nucl_acid_amp_date</a></td>
<td>the date of the nucleoacid amplification</td>
</tr>
<tr>
<td><a href="nucl_acid_ext/">nucl_acid_ext</a></td>
<td>Link to a reference or kit that describes the recovery of nucleic acids from ...</td>
</tr>
<tr>
<td><a href="nucl_acid_ext_date/">nucl_acid_ext_date</a></td>
<td>the date of the nucleoacid extraction</td>
</tr>
<tr>
<td><a href="panel_id/">panel_id</a></td>
<td>the index into the panel_info list</td>
</tr>
<tr>
<td><a href="panel_info/">panel_info</a></td>
<td>a list of info on the panels</td>
</tr>
<tr>
<td><a href="panel_name/">panel_name</a></td>
<td>a name for the panel</td>
</tr>
<tr>
<td><a href="panel_targets/">panel_targets</a></td>
<td>a list of the target indexes in the target_info list</td>
</tr>
<tr>
<td><a href="parasite_density/">parasite_density</a></td>
<td>the density in microliters</td>
</tr>
<tr>
<td><a href="pcr_cond/">pcr_cond</a></td>
<td>the method/conditions for PCR, List PCR cycles used to amplify the target</td>
</tr>
<tr>
<td><a href="plate_col/">plate_col</a></td>
<td>the column the specimen was in</td>
</tr>
<tr>
<td><a href="plate_name/">plate_name</a></td>
<td>a name of plate the specimen was in</td>
</tr>
<tr>
<td><a href="plate_row/">plate_row</a></td>
<td>the row the specimen was in</td>
</tr>
<tr>
<td><a href="pmo_header/">pmo_header</a></td>
<td>the PMO information for this file including version etc</td>
</tr>
<tr>
<td><a href="pmo_version/">pmo_version</a></td>
<td>the version of the PMO file, should be in the format of v[MAJOR]</td>
</tr>
<tr>
<td><a href="program/">program</a></td>
<td>name of the program used for this portion of the pipeline</td>
</tr>
<tr>
<td><a href="program_description/">program_description</a></td>
<td>a short description of what this method does</td>
</tr>
<tr>
<td><a href="program_name/">program_name</a></td>
<td>the name of the program</td>
</tr>
<tr>
<td><a href="program_version/">program_version</a></td>
<td>the version of generation method, should be in the format of v[MAJOR]</td>
</tr>
<tr>
<td><a href="project_collector_chief_scientist/">project_collector_chief_scientist</a></td>
<td>can be collection of names separated by a semicolon if multiple people involv...</td>
</tr>
<tr>
<td><a href="project_contributors/">project_contributors</a></td>
<td>a list of collaborators who contributed to this project</td>
</tr>
<tr>
<td><a href="project_description/">project_description</a></td>
<td>a short description of the project</td>
</tr>
<tr>
<td><a href="project_id/">project_id</a></td>
<td>the index into the project_info list</td>
</tr>
<tr>
<td><a href="project_info/">project_info</a></td>
<td>the information about the projects stored in this PMO</td>
</tr>
<tr>
<td><a href="project_name/">project_name</a></td>
<td>a name for the project, should be unique if multiple projects listed</td>
</tr>
<tr>
<td><a href="project_type/">project_type</a></td>
<td>the type of project conducted, e</td>
</tr>
<tr>
<td><a href="pseudocigar/">pseudocigar</a></td>
<td>the pseudocigar of the haplotype</td>
</tr>
<tr>
<td><a href="qpcr_parasite_density_info/">qpcr_parasite_density_info</a></td>
<td>qpcr parasite density measurement</td>
</tr>
<tr>
<td><a href="quality/">quality</a></td>
<td>the ansi fastq per base quality score for this sequence, this is optional</td>
</tr>
<tr>
<td><a href="reaction_name/">reaction_name</a></td>
<td>a name for this reaction</td>
</tr>
<tr>
<td><a href="reactions/">reactions</a></td>
<td>a list of 1 or more reactions that this panel contains, each reactions list t...</td>
</tr>
<tr>
<td><a href="read_count/">read_count</a></td>
<td>the read counts</td>
</tr>
<tr>
<td><a href="read_counts_by_experimental_sample_by_stage/">read_counts_by_experimental_sample_by_stage</a></td>
<td>a list by experiment_sample for the counts at each stage</td>
</tr>
<tr>
<td><a href="read_counts_by_stage/">read_counts_by_stage</a></td>
<td>the read counts for different stages of the pipeline</td>
</tr>
<tr>
<td><a href="read_counts_for_targets/">read_counts_for_targets</a></td>
<td>a list of counts by stage for a target</td>
</tr>
<tr>
<td><a href="reads/">reads</a></td>
<td>the read count associated with this microhaplotype</td>
</tr>
<tr>
<td><a href="ref_seq/">ref_seq</a></td>
<td>the reference sequence of this genomic location</td>
</tr>
<tr>
<td><a href="replacement_size/">replacement_size</a></td>
<td>the size of replacement mask</td>
</tr>
<tr>
<td><a href="representative_microhaplotypes/">representative_microhaplotypes</a></td>
<td>a list of the information on the representative microhaplotypes</td>
</tr>
<tr>
<td><a href="reverse_primer/">reverse_primer</a></td>
<td>the reverse primer associated with this target</td>
</tr>
<tr>
<td><a href="run_date/">run_date</a></td>
<td>the date when the run was done, should be YYYY-MM-DD</td>
</tr>
<tr>
<td><a href="seq/">seq</a></td>
<td>the DNA sequence</td>
</tr>
<tr>
<td><a href="seq_center/">seq_center</a></td>
<td>Name of facility where sequencing was performed (lab, core facility, or compa...</td>
</tr>
<tr>
<td><a href="seq_date/">seq_date</a></td>
<td>the date of sequencing, should be YYYY-MM or YYYY-MM-DD</td>
</tr>
<tr>
<td><a href="seq_instrument_model/">seq_instrument_model</a></td>
<td>the sequencing instrument model used to sequence the run, e</td>
</tr>
<tr>
<td><a href="seq_platform/">seq_platform</a></td>
<td>the sequencing technology used to sequence the run, e</td>
</tr>
<tr>
<td><a href="seq_segment_size/">seq_segment_size</a></td>
<td>the size of the masking</td>
</tr>
<tr>
<td><a href="seq_start/">seq_start</a></td>
<td>the start of the masking</td>
</tr>
<tr>
<td><a href="sequencing_info/">sequencing_info</a></td>
<td>a list of sequencing info for this project</td>
</tr>
<tr>
<td><a href="sequencing_info_id/">sequencing_info_id</a></td>
<td>the index into the sequencing_info list</td>
</tr>
<tr>
<td><a href="sequencing_info_name/">sequencing_info_name</a></td>
<td>a name of for the sequencing done, e</td>
</tr>
<tr>
<td><a href="specimen_collect_device/">specimen_collect_device</a></td>
<td>the way the specimen was collected, e</td>
</tr>
<tr>
<td><a href="specimen_comments/">specimen_comments</a></td>
<td>any additional comments about the specimen</td>
</tr>
<tr>
<td><a href="specimen_id/">specimen_id</a></td>
<td>the index into the specimen_info list</td>
</tr>
<tr>
<td><a href="specimen_info/">specimen_info</a></td>
<td>a list of all the specimens within this project</td>
</tr>
<tr>
<td><a href="specimen_name/">specimen_name</a></td>
<td>an identifier for the specimen, should be unique within this sample set</td>
</tr>
<tr>
<td><a href="specimen_store_loc/">specimen_store_loc</a></td>
<td>the specimen store site, address or facility name</td>
</tr>
<tr>
<td><a href="specimen_taxon_id/">specimen_taxon_id</a></td>
<td>the NCBI taxonomy number of the organism in specimen, can list multiple if a ...</td>
</tr>
<tr>
<td><a href="specimen_type/">specimen_type</a></td>
<td>what type of specimen this is, e</td>
</tr>
<tr>
<td><a href="stage/">stage</a></td>
<td>the stage of the pipeline, e</td>
</tr>
<tr>
<td><a href="stages/">stages</a></td>
<td>the read counts by each stage</td>
</tr>
<tr>
<td><a href="start/">start</a></td>
<td>the start of the location, 0-based positioning</td>
</tr>
<tr>
<td><a href="storage_plate_info/">storage_plate_info</a></td>
<td>plate location of where specimen is stored if stored in a plate</td>
</tr>
<tr>
<td><a href="strand/">strand</a></td>
<td>which strand the location is, either + for plus strand or - for negative stra...</td>
</tr>
<tr>
<td><a href="target_attributes/">target_attributes</a></td>
<td>a list of classification type for the primer target</td>
</tr>
<tr>
<td><a href="target_id/">target_id</a></td>
<td>the index into the target_info list</td>
</tr>
<tr>
<td><a href="target_info/">target_info</a></td>
<td>a list of info on the targets</td>
</tr>
<tr>
<td><a href="target_name/">target_name</a></td>
<td>an identifier for this target</td>
</tr>
<tr>
<td><a href="target_results/">target_results</a></td>
<td>a list of the microhaplotypes detected for a list of targets</td>
</tr>
<tr>
<td><a href="targeted_genomes/">targeted_genomes</a></td>
<td>a list of genomes that the targets in TargetInfo refer to</td>
</tr>
<tr>
<td><a href="targets/">targets</a></td>
<td>a list of the microhaplotype for each targets</td>
</tr>
<tr>
<td><a href="taxon_id/">taxon_id</a></td>
<td>the NCBI taxonomy number</td>
</tr>
<tr>
<td><a href="total_raw_count/">total_raw_count</a></td>
<td>the raw counts off the sequencing machine that a sample began with</td>
</tr>
<tr>
<td><a href="travel_out_six_month/">travel_out_six_month</a></td>
<td>Specification of the countries travelled in the last six months; can include ...</td>
</tr>
<tr>
<td><a href="umis/">umis</a></td>
<td>the unique molecular identifier (umi) count associated with this microhaploty...</td>
</tr>
<tr>
<td><a href="url/">url</a></td>
<td>a link to the where this genome file could be downloaded</td>
</tr>
</tbody>
</table>
<h2 id="enumerations">Enumerations</h2>
<table>
<thead>
<tr>
<th>Enumeration</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="types">Types</h2>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="Boolean/">Boolean</a></td>
<td>A binary (true or false) value</td>
</tr>
<tr>
<td><a href="Curie/">Curie</a></td>
<td>a compact URI</td>
</tr>
<tr>
<td><a href="Date/">Date</a></td>
<td>a date (year, month and day) in an idealized calendar</td>
</tr>
<tr>
<td><a href="DateOrDatetime/">DateOrDatetime</a></td>
<td>Either a date or a datetime</td>
</tr>
<tr>
<td><a href="Datetime/">Datetime</a></td>
<td>The combination of a date and time</td>
</tr>
<tr>
<td><a href="Decimal/">Decimal</a></td>
<td>A real number with arbitrary precision that conforms to the xsd:decimal speci...</td>
</tr>
<tr>
<td><a href="Double/">Double</a></td>
<td>A real number that conforms to the xsd:double specification</td>
</tr>
<tr>
<td><a href="Float/">Float</a></td>
<td>A real number that conforms to the xsd:float specification</td>
</tr>
<tr>
<td><a href="Integer/">Integer</a></td>
<td>An integer</td>
</tr>
<tr>
<td><a href="Jsonpath/">Jsonpath</a></td>
<td>A string encoding a JSON Path</td>
</tr>
<tr>
<td><a href="Jsonpointer/">Jsonpointer</a></td>
<td>A string encoding a JSON Pointer</td>
</tr>
<tr>
<td><a href="Ncname/">Ncname</a></td>
<td>Prefix part of CURIE</td>
</tr>
<tr>
<td><a href="Nodeidentifier/">Nodeidentifier</a></td>
<td>A URI, CURIE or BNODE that represents a node in a model</td>
</tr>
<tr>
<td><a href="Objectidentifier/">Objectidentifier</a></td>
<td>A URI or CURIE that represents an object in the model</td>
</tr>
<tr>
<td><a href="Sparqlpath/">Sparqlpath</a></td>
<td>A string encoding a SPARQL Property Path</td>
</tr>
<tr>
<td><a href="String/">String</a></td>
<td>A character string</td>
</tr>
<tr>
<td><a href="Time/">Time</a></td>
<td>A time object represents a (local) time of day, independent of any particular...</td>
</tr>
<tr>
<td><a href="Uri/">Uri</a></td>
<td>a complete URI</td>
</tr>
<tr>
<td><a href="Uriorcurie/">Uriorcurie</a></td>
<td>a URI or a CURIE</td>
</tr>
</tbody>
</table>
<h2 id="subsets">Subsets</h2>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": ".", "features": ["content.tabs.link"], "search": "assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="assets/javascripts/bundle.56ea9cef.min.js"></script>
      
        <script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
      
        <script src="javascripts/tablesort.js"></script>
      
    
  </body>
</html>