
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="https://PlasmoGenEpi.github.io/portable-microhaplotype-object/SequencingInfo/" rel="canonical"/>
<link href="../assets/images/favicon.png" rel="icon"/>
<meta content="mkdocs-1.6.0, mkdocs-material-8.5.10" name="generator"/>
<title>Class: SequencingInfo - portable-microhaplotype-object</title>
<link href="../assets/stylesheets/main.975780f9.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.2505c338.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<link href="../css/ansi-colours.css" rel="stylesheet"/>
<link href="../css/jupyter-cells.css" rel="stylesheet"/>
<link href="../css/pandas-dataframe.css" rel="stylesheet"/>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
</head>
<body data-md-color-accent="None" data-md-color-primary="None" data-md-color-scheme="default" dir="ltr">
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#class-sequencinginfo">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="portable-microhaplotype-object" class="md-header__button md-logo" data-md-component="logo" href=".." title="portable-microhaplotype-object">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            portable-microhaplotype-object
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Class: SequencingInfo
            
          </span>
</div>
</div>
</div>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/PlasmoGenEpi/portable-microhaplotype-object" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="portable-microhaplotype-object" class="md-nav__button md-logo" data-md-component="logo" href=".." title="portable-microhaplotype-object">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"></path></svg>
</a>
    portable-microhaplotype-object
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/PlasmoGenEpi/portable-microhaplotype-object" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
        Index
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../about/">
        About
      </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#slots">
    Slots
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usages">
    Usages
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#identifier-and-mapping-information">
    Identifier and Mapping Information
  </a>
<nav aria-label="Identifier and Mapping Information" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#schema-source">
    Schema Source
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#mappings">
    Mappings
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#linkml-source">
    LinkML Source
  </a>
<nav aria-label="LinkML Source" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#direct">
    Direct
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#induced">
    Induced
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<a class="md-content__button md-icon" href="https://github.com/PlasmoGenEpi/portable-microhaplotype-object/edit/master/docs/SequencingInfo.md" title="Edit this page">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg>
</a>
<h1 id="class-sequencinginfo">Class: SequencingInfo</h1>
<p><em>Information on sequencing info</em></p>
<p>URI: <a href="https://plasmogenepi.github.io/portable-microhaplotype-object/SequencingInfo">portable_microhaplotype_object:SequencingInfo</a></p>
<div class="mermaid"> classDiagram
    class SequencingInfo
    click SequencingInfo href "../SequencingInfo"
      SequencingInfo : library_kit

      SequencingInfo : library_layout

      SequencingInfo : library_screen

      SequencingInfo : library_selection

      SequencingInfo : library_source

      SequencingInfo : library_strategy

      SequencingInfo : nucl_acid_amp

      SequencingInfo : nucl_acid_amp_date

      SequencingInfo : nucl_acid_ext

      SequencingInfo : nucl_acid_ext_date

      SequencingInfo : pcr_cond

      SequencingInfo : seq_center

      SequencingInfo : seq_date

      SequencingInfo : seq_instrument_model

      SequencingInfo : seq_platform

      SequencingInfo : sequencing_info_name


</div>
<!-- no inheritance hierarchy -->
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Cardinality and Range</th>
<th>Description</th>
<th>Inheritance</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../sequencing_info_name/">sequencing_info_name</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>a name of for the sequencing done, e</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../seq_platform/">seq_platform</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>the sequencing technology used to sequence the run, e</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../seq_instrument_model/">seq_instrument_model</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>the sequencing instrument model used to sequence the run, e</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../seq_date/">seq_date</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>the date of sequencing, should be YYYY-MM or YYYY-MM-DD</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../nucl_acid_ext/">nucl_acid_ext</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Link to a reference or kit that describes the recovery of nucleic acids from ...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../nucl_acid_amp/">nucl_acid_amp</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Link to a reference or kit that describes the enzymatic amplification of nucl...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../nucl_acid_ext_date/">nucl_acid_ext_date</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>the date of the nucleoacid extraction</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../nucl_acid_amp_date/">nucl_acid_amp_date</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>the date of the nucleoacid amplification</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../pcr_cond/">pcr_cond</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>the method/conditions for PCR, List PCR cycles used to amplify the target</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../library_screen/">library_screen</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Describe enrichment, screening, or normalization methods applied during ampli...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../library_kit/">library_kit</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Name, version, and applicable cell or cycle numbers for the kit used to prepa...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../library_layout/">library_layout</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Specify the configuration of reads, e</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../library_strategy/">library_strategy</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>what the nuceloacid sequencing/amplification strategy was (common names are A...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../library_source/">library_source</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>Source of amplification material (common names GENOMIC, TRANSCRIPTOMIC)</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../library_selection/">library_selection</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>how amplification was done (common are PCR=Source material was selected by de...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../seq_center/">seq_center</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Name of facility where sequencing was performed (lab, core facility, or compa...</td>
<td>direct</td>
</tr>
</tbody>
</table>
<h2 id="usages">Usages</h2>
<table>
<thead>
<tr>
<th>used by</th>
<th>used in</th>
<th>type</th>
<th>used</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../PortableMicrohaplotypeObject/">PortableMicrohaplotypeObject</a></td>
<td><a href="../sequencing_info/">sequencing_info</a></td>
<td>range</td>
<td><a href="./">SequencingInfo</a></td>
</tr>
</tbody>
</table>
<h2 id="identifier-and-mapping-information">Identifier and Mapping Information</h2>
<h3 id="schema-source">Schema Source</h3>
<ul>
<li>from schema: https://plasmogenepi.github.io/portable-microhaplotype-object</li>
</ul>
<h2 id="mappings">Mappings</h2>
<table>
<thead>
<tr>
<th>Mapping Type</th>
<th>Mapped Value</th>
</tr>
</thead>
<tbody>
<tr>
<td>self</td>
<td>portable_microhaplotype_object:SequencingInfo</td>
</tr>
<tr>
<td>native</td>
<td>portable_microhaplotype_object:SequencingInfo</td>
</tr>
</tbody>
</table>
<h2 id="linkml-source">LinkML Source</h2>
<!-- TODO: investigate https://stackoverflow.com/questions/37606292/how-to-create-tabbed-code-blocks-in-mkdocs-or-sphinx -->
<h3 id="direct">Direct</h3>
<details>
<pre><code class="language-yaml">name: SequencingInfo
description: Information on sequencing info
from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
attributes:
  sequencing_info_name:
    name: sequencing_info_name
    description: a name of for the sequencing done, e.g. batch1
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  seq_platform:
    name: seq_platform
    description: the sequencing technology used to sequence the run, e.g. ILLUMINA,
      NANOPORE, PACBIO
    examples:
    - value: ILLUMINA
    - value: PACBIO
    - value: NANOPORE
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  seq_instrument_model:
    name: seq_instrument_model
    description: the sequencing instrument model used to sequence the run, e.g. NextSeq
      2000, MinION, Revio
    examples:
    - value: NextSeq 2000
    - value: MinION
    - value: Revio
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  seq_date:
    name: seq_date
    description: the date of sequencing, should be YYYY-MM or YYYY-MM-DD
    examples:
    - value: '2024-03-10'
    - value: 2024-03
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: \d{4}-(?:0[1-9]|1[0-2])(?:-(?:0[1-9]|[12][0-9]|3[01]))?
  nucl_acid_ext:
    name: nucl_acid_ext
    description: Link to a reference or kit that describes the recovery of nucleic
      acids from the sample
    examples:
    - value: https://www.paragongenomics.com/targeted-sequencing/amplicon-sequencing/cleanplex-ngs-amplicon-sequencing/
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$
  nucl_acid_amp:
    name: nucl_acid_amp
    description: Link to a reference or kit that describes the enzymatic amplification
      of nucleic acids
    examples:
    - value: https://www.paragongenomics.com/targeted-sequencing/amplicon-sequencing/cleanplex-ngs-amplicon-sequencing/
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$
  nucl_acid_ext_date:
    name: nucl_acid_ext_date
    description: the date of the nucleoacid extraction
    examples:
    - value: '2024-03-10'
    - value: 2024-03
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: \d{4}-(?:0[1-9]|1[0-2])(?:-(?:0[1-9]|[12][0-9]|3[01]))?
  nucl_acid_amp_date:
    name: nucl_acid_amp_date
    description: the date of the nucleoacid amplification
    examples:
    - value: '2024-03-10'
    - value: 2024-03
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: \d{4}-(?:0[1-9]|1[0-2])(?:-(?:0[1-9]|[12][0-9]|3[01]))?
  pcr_cond:
    name: pcr_cond
    description: the method/conditions for PCR, List PCR cycles used to amplify the
      target
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^[A-z-._0-9\(\),\/\ ]+$
  library_screen:
    name: library_screen
    description: Describe enrichment, screening, or normalization methods applied
      during amplification or library preparation, e.g. size selection 390bp, diluted
      to 1 ng DNA/sample
    examples:
    - value: size selection 390bp, diluted to 1 ng DNA/sample
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^[A-z-._0-9\(\),\/\ ]+$
  library_kit:
    name: library_kit
    description: Name, version, and applicable cell or cycle numbers for the kit used
      to prepare libraries and load cells or chips for sequencing. If possible, include
      a part number, e.g. MiSeq Reagent Kit v3 (150-cycle), MS-102-3001
    examples:
    - value: MiSeq Reagent Kit v3 (150-cycle), MS-102-3001
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^[A-z-._0-9\(\),\/\ ]+$
  library_layout:
    name: library_layout
    description: Specify the configuration of reads, e.g. paired-end, single
    examples:
    - value: paired-end
    - value: single
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  library_strategy:
    name: library_strategy
    description: what the nuceloacid sequencing/amplification strategy was (common
      names are AMPLICON, WGS)
    examples:
    - value: AMPLICON
    - value: WGS
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  library_source:
    name: library_source
    description: Source of amplification material (common names GENOMIC, TRANSCRIPTOMIC)
    examples:
    - value: GENOMIC
    - value: TRANSCRIPTOMIC
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  library_selection:
    name: library_selection
    description: how amplification was done (common are PCR=Source material was selected
      by designed primers, RANDOM =Random selection by shearing or other method)
    examples:
    - value: PCR
    - value: RANDOM
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  seq_center:
    name: seq_center
    description: Name of facility where sequencing was performed (lab, core facility,
      or company)
    examples:
    - value: UCSF sequencing core
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^[A-z-._0-9\(\),\/\ ]+$

</code></pre>
</details>
<h3 id="induced">Induced</h3>
<details>
<pre><code class="language-yaml">name: SequencingInfo
description: Information on sequencing info
from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
attributes:
  sequencing_info_name:
    name: sequencing_info_name
    description: a name of for the sequencing done, e.g. batch1
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: sequencing_info_name
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  seq_platform:
    name: seq_platform
    description: the sequencing technology used to sequence the run, e.g. ILLUMINA,
      NANOPORE, PACBIO
    examples:
    - value: ILLUMINA
    - value: PACBIO
    - value: NANOPORE
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: seq_platform
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  seq_instrument_model:
    name: seq_instrument_model
    description: the sequencing instrument model used to sequence the run, e.g. NextSeq
      2000, MinION, Revio
    examples:
    - value: NextSeq 2000
    - value: MinION
    - value: Revio
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: seq_instrument_model
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  seq_date:
    name: seq_date
    description: the date of sequencing, should be YYYY-MM or YYYY-MM-DD
    examples:
    - value: '2024-03-10'
    - value: 2024-03
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: seq_date
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: \d{4}-(?:0[1-9]|1[0-2])(?:-(?:0[1-9]|[12][0-9]|3[01]))?
  nucl_acid_ext:
    name: nucl_acid_ext
    description: Link to a reference or kit that describes the recovery of nucleic
      acids from the sample
    examples:
    - value: https://www.paragongenomics.com/targeted-sequencing/amplicon-sequencing/cleanplex-ngs-amplicon-sequencing/
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: nucl_acid_ext
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$
  nucl_acid_amp:
    name: nucl_acid_amp
    description: Link to a reference or kit that describes the enzymatic amplification
      of nucleic acids
    examples:
    - value: https://www.paragongenomics.com/targeted-sequencing/amplicon-sequencing/cleanplex-ngs-amplicon-sequencing/
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: nucl_acid_amp
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$
  nucl_acid_ext_date:
    name: nucl_acid_ext_date
    description: the date of the nucleoacid extraction
    examples:
    - value: '2024-03-10'
    - value: 2024-03
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: nucl_acid_ext_date
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: \d{4}-(?:0[1-9]|1[0-2])(?:-(?:0[1-9]|[12][0-9]|3[01]))?
  nucl_acid_amp_date:
    name: nucl_acid_amp_date
    description: the date of the nucleoacid amplification
    examples:
    - value: '2024-03-10'
    - value: 2024-03
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: nucl_acid_amp_date
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: \d{4}-(?:0[1-9]|1[0-2])(?:-(?:0[1-9]|[12][0-9]|3[01]))?
  pcr_cond:
    name: pcr_cond
    description: the method/conditions for PCR, List PCR cycles used to amplify the
      target
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: pcr_cond
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^[A-z-._0-9\(\),\/\ ]+$
  library_screen:
    name: library_screen
    description: Describe enrichment, screening, or normalization methods applied
      during amplification or library preparation, e.g. size selection 390bp, diluted
      to 1 ng DNA/sample
    examples:
    - value: size selection 390bp, diluted to 1 ng DNA/sample
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: library_screen
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^[A-z-._0-9\(\),\/\ ]+$
  library_kit:
    name: library_kit
    description: Name, version, and applicable cell or cycle numbers for the kit used
      to prepare libraries and load cells or chips for sequencing. If possible, include
      a part number, e.g. MiSeq Reagent Kit v3 (150-cycle), MS-102-3001
    examples:
    - value: MiSeq Reagent Kit v3 (150-cycle), MS-102-3001
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: library_kit
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^[A-z-._0-9\(\),\/\ ]+$
  library_layout:
    name: library_layout
    description: Specify the configuration of reads, e.g. paired-end, single
    examples:
    - value: paired-end
    - value: single
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: library_layout
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  library_strategy:
    name: library_strategy
    description: what the nuceloacid sequencing/amplification strategy was (common
      names are AMPLICON, WGS)
    examples:
    - value: AMPLICON
    - value: WGS
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: library_strategy
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  library_source:
    name: library_source
    description: Source of amplification material (common names GENOMIC, TRANSCRIPTOMIC)
    examples:
    - value: GENOMIC
    - value: TRANSCRIPTOMIC
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: library_source
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  library_selection:
    name: library_selection
    description: how amplification was done (common are PCR=Source material was selected
      by designed primers, RANDOM =Random selection by shearing or other method)
    examples:
    - value: PCR
    - value: RANDOM
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: library_selection
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: true
    pattern: ^[A-z-._0-9 ]+$
  seq_center:
    name: seq_center
    description: Name of facility where sequencing was performed (lab, core facility,
      or company)
    examples:
    - value: UCSF sequencing core
    from_schema: https://plasmogenepi.github.io/portable-microhaplotype-object
    rank: 1000
    alias: seq_center
    owner: SequencingInfo
    domain_of:
    - SequencingInfo
    range: string
    required: false
    pattern: ^[A-z-._0-9\(\),\/\ ]+$

</code></pre>
</details>
</article>
</div>
<script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>
</div>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
<script src="../assets/javascripts/bundle.5a2dcb6a.min.js"></script>
<script src="https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js"></script>
<script src="../javascripts/tablesort.js"></script>
<script type="module">import mermaid from "https://unpkg.com/mermaid@10.4.0/dist/mermaid.esm.min.mjs";
mermaid.initialize({});</script><script type="module">import mermaid from "https://unpkg.com/mermaid@10.9.0/dist/mermaid.esm.min.mjs";
mermaid.initialize({});</script></body>
</html>